hypothesis-python/src/hypothesis/core.py

```diff
@@ -1101,24 +1116,13 @@ def wrapped_test(*arguments, **kwargs):
                     )
                     raise the_error_hypothesis_found
 
-        def fuzz_one_input(
-            buffer: Union[bytes, bytearray, memoryview, BinaryIO]
-        ) -> Optional[bytes]:
-            """Run the test as a fuzz target, driven with the `buffer` of bytes.
-
-            Returns None if buffer invalid for the strategy, canonical pruned
-            bytes if the buffer was valid, and leaves raised exceptions alone.
-
-            Note: this feature is experimental and may change or be removed.
-            """
-            if isinstance(buffer, io.IOBase):
-                buffer = buffer.read()
-            if isinstance(buffer, (bytearray, memoryview)):
-                buffer = bytes(buffer)
-            assert isinstance(buffer, bytes)
-
+        def _get_fuzz_target() -> Callable[
+            [Union[bytes, bytearray, memoryview, BinaryIO]], Optional[bytes]
+        ]:
             test = wrapped_test.hypothesis.inner_test
```

Comment:
It'd be nice to have a comment here or elsewhere that we're adding the indirection to improve performance.